golden/runner.py

```
@@ -323,6 +325,120 @@ def _execute_via_runner(
+    L2 single-chip only: delegates to :func:`pypto.runtime.benchmark`, which
+    opens one :class:`~pypto.runtime.ChipWorker`, registers *compiled* once, and
+    reads each launch's on-NPU ``device_wall_us`` from the runtime's
+    ``[STRACE]`` markers (simpler PR #1177). Args are reordered to the
+    orchestration parameter order exactly as :func:`_execute_via_runner` does.
+    Returns the :class:`~pypto.runtime.BenchmarkStats`, or ``None`` when the
+    runtime emits no markers (built without ``SIMPLER_PROFILING``).
+    """
+    from pypto.runtime import benchmark
+
+    def _effective_us(inv: Any) -> float | None:
+        """One launch's Effective window (µs): the orch∪sched merged span.
```

Ruff (0.15.20)

[warning] Docstring contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF002)